pkg/notify/notify.go

```diff
@@ -135,11 +97,55 @@ func (n *Notifier) Close() {
 	n.mu.Lock()
 	defer n.mu.Unlock()
 	for _, receiver := range n.receivers {
-		if receiver.rec.ticker != nil {
-			receiver.rec.ticker.Stop()
-		}
-		close(receiver.rec.closeCh)
+		receiver.rec.close()
 	}
 	n.receivers = nil
 	n.closed = true
 }
+
+// Receiver is a receiver of notifier, including the receiver channel and stop receiver function.
+type Receiver struct {
+	C       <-chan struct{}
+	c       chan struct{}
+	Stop    func()
+	ticker  *time.Ticker
+	closeCh chan struct{}
+}
+
+// returns true if the receiverCh should be closed
+func (r *Receiver) signalNonBlocking() bool {
+	select {
+	case <-r.closeCh:
+		return true
+	case r.c <- struct{}{}:
+	default:
+	}
+	return false
+}
+
+func (r *Receiver) signalTickLoop() {
+	go func() {
+	loop:
+		for {
+			select {
+			case <-r.closeCh:
+				break loop
+			case <-r.ticker.C:
+			}
+			exit := r.signalNonBlocking()
+			if exit {
+				break loop
+			}
+		}
+		close(r.c)
```

Comment:
Can add some comments here to explain why we `close(r.c)` here instead doing it in `receiver.close()`

Reply:
[added, PTAL](https://github.com/pingcap/tiflow/pull/7077/files#diff-d978136a23c5c118a3dc8719ba1abeeb49bb025f44d4d09555602128ab4f79a9R146-R147)